### PR TITLE
dep: bump up caddy version to 2.7.3

### PR DESCRIPTION
**Pull request by evict**
This PR makes sure we use a later version of Caddy, with less vulnerabilities. We move from version 2.4.5 to 2.7.3.

## Test plan
Tested the changes locally using `sg start`, confirms it's still serving SSL with 2.7.3.

### CODE REVIEW

**Review by willdollman — APPROVED** [review]

File: dev/caddy.sh [hunk 1]
```diff
@@ -6,7 +6,7 @@ pushd "$(dirname "${BASH_SOURCE[0]}")/.." >/dev/null
 
 mkdir -p .bin
 
-version="2.4.5"
+version="2.7.2"
```
Comment (burmudar) [inline dev/caddy.sh:9]:
```suggestion
version="2.7.3"
```
> Do not use this release, use [v2.7.3](https://github.com/caddyserver/caddy/releases/tag/v2.7.3) instead which contains several hot fixes.

**Review by burmudar — COMMENTED** [review]
2.7.3 got released over the weekend and the official recommendation is to use that rather than 2.7.2

**Review by burmudar — CHANGES_REQUESTED** [review]
See previous comment

File: sg.config.yaml [hunk 2]
```diff
@@ -331,7 +331,7 @@ commands:
     cmd: .bin/caddy_${CADDY_VERSION} run --watch --config=dev/Caddyfile
     install_func: installCaddy
     env:
-      CADDY_VERSION: 2.4.5
+      CADDY_VERSION: 2.7.2
```
Comment (burmudar) [inline sg.config.yaml:334]:
```suggestion
      CADDY_VERSION: 2.7.3
```

**Review by burmudar — APPROVED** [review]

**Comment by willdollman** [comment]
Waiting on caddy 2.7.3 docker images to be released

**Comment by willdollman** [comment]
Caddy 2.7.3 released https://hub.docker.com/layers/library/caddy/2.7.3/images/sha256-f9824933254e3e43e0508670ee9bdcde704621017e95119a05317383b1878f4f?context=explore

**Pull request merged**